apps/condo/domains/organization/tasks/syncSbbolTransactions.js

```diff
@@ -46,18 +54,41 @@ async function syncSbbolTransactions (date, userId = '', organization = {}) {
 
         if (employee) {
             await requestTransactions({
-                date,
+                dateInterval,
                 userId,
                 organization: get(employee, 'organization'),
             })
         }
     }
 }
 
+/**
+ * Synchronizes SBBOL transaction data with data in the system
+ * @returns {Promise<void|Transaction[]>}
+ */
+async function syncSbbolTransactionsBankSyncTask (dateInterval, userId, organization, taskId) {
+    const { keystone: context } = await getSchemaCtx('User')
+
+    if (!dateInterval || !userId || !organization || !taskId) {
+        await bankSyncTaskUpdateStatus(context, taskId, BANK_SYNC_TASK_STATUS.ERROR)
+        throw new Error('Missing setup args')
+    }
+
+    try {
+        await requestTransactions({ dateInterval, userId, organization })
+    } catch (e) {
+        await bankSyncTaskUpdateStatus(context, taskId, BANK_SYNC_TASK_STATUS.ERROR)
+        throw new Error(`Cannot requestTransactions. ${e}`)
+    }
+
+    await bankSyncTaskUpdateStatus(context, taskId, BANK_SYNC_TASK_STATUS.COMPLETED)
```

Comment:
Is there any real way to calculate task progress?

Reply:
correctly no. If you do it in a large cell, you can, but I don’t have time for this, the release branch is being created today. Therefore, we will simply display the status

Reply:
You can calculate the progress based on number of synced days from interval.

Reply:
as I said, these are calculations in a large cell